api/app/models/fetch/predictions.py

```diff
@@ -192,6 +190,68 @@ async def _fetch_model_predictions_by_station_codes(model: ModelEnum, station_co
 
 
 async def fetch_model_predictions(model: ModelEnum, station_codes: List[int]):
-    """ Fetch 10 day global model weather predictions for a given station."""
+    """ Fetch model weather predictions for a given list of stations and a given model. """
     # Fetch predictions from the database.
     return await _fetch_model_predictions_by_station_codes(model, station_codes)
+
+
+async def _fetch_most_recent_historic_predictions_by_station_codes(model: ModelEnum, station_codes: List[int]) -> List[WeatherModelPrediction]:
+    """ Fetch the most recent historic model predictions from database based on each station's coordinates. """
+    # construct temporary helper dictionary of relevant weather stations
+    stations_dict = {}
```

Comment:
This 1 line of code would replace those 4:
```suggestion
    stations = {station.code : station for station in await get_stations_by_codes(station_codes)}
```